addons/animated-thumb/userscript.js

```diff
@@ -5,33 +5,28 @@ export default async function ({ addon, global, console, msg }) {
     let nav = await addon.tab.waitForElement("[class^='menu-bar_main-menu']", {
       markAsSeen: true,
     });
-    if (!document.querySelector("[class^='author-info_username-line']")) {
-      let setthumb = document.createElement("div");
-      setthumb.classList.add(addon.tab.scratchClass("menu-bar_menu-bar-item"));
-      setthumb.title = msg("added-by");
-      let thumbinner = document.createElement("span");
-      thumbinner.setAttribute(
-        "class",
-        addon.tab.scratchClass(
-          "button_outlined-button",
-          "menu-bar_menu-bar-button",
-          "community-button_community-button"
-        )
-      );
-      thumbinner.setAttribute("role", "button");
-      setthumb.append(thumbinner);
-      let thumbcontent = document.createElement("div");
-      setthumb.classList.add(addon.tab.scratchClass("button_content"));
-      thumbinner.append(thumbcontent);
-      let thumbspan = document.createElement("span");
-      thumbspan.innerText = msg("set-thumbnail");
-      thumbcontent.append(thumbspan);
-      nav.append(setthumb);
-      setthumb.addEventListener("click", function (e) {
-        const setter = new ThumbSetter((key) => msg(`/${key}`));
-        setter.addFileInput();
-        setter.showInput();
-      });
-    }
+    if (!document.querySelectorAll("[class*='project-title-input_title-field']")) return;
```

Comment:
why not check `addon.auth.isLoggedIn`?

Reply:
@RedGuy12 That would be a better idea because Scratch sometimes needs a while to check if the user is logged in or not.

Reply:
This previously used whether the project says "by username" to decide if the button should be added or not, which probably uses the same API request as checking whether the editor should have editable title 🤔